core/lib/dal/src/blocks_dal.rs

```diff
@@ -1021,12 +1022,14 @@ impl BlocksDal<'_, '_> {
                 AND protocol_versions.bootloader_code_hash = $1 AND protocol_versions.default_account_code_hash = $2 \
                 AND commitment IS NOT NULL \
                 AND (protocol_versions.id = $3 OR protocol_versions.upgrade_tx_hash IS NULL) \
-                AND events_queue_commitment IS NOT NULL \
-                AND bootloader_initial_content_commitment IS NOT NULL
-            ORDER BY number LIMIT $4",
+                AND (events_queue_commitment IS NOT NULL AND bootloader_initial_content_commitment IS NOT NULL
+                    OR $4 = FALSE
+                )
+            ORDER BY number LIMIT $5",
```

Comment:
TBH, i think it's better separate it to two methods...

Reply:
Ok, I agree that DAL method whose behavior depends on protocol version is too much